Generate simulated request for testing

terezka opened the pull request:
TODO:
- [ ] Ensure this works with monorepo lts version (12.26)

terezka committed:
Straighten type signatures 😈

terezka committed:
Generate simulated http request

stoeffel committed:
update stack.yaml

location under packages is deprecated

stoeffel committed:
fix specs

stoeffel commented:
@terezka I based this on #11 , but actually that might not work on travis, because we still install an old version of stack there.

stoeffel commented:
no, might be something else.
seems to have problems installing elm-export

stoeffel committed:
use https instead of ssh

stoeffel committed:
upgrade to latest elm-export

stoeffel committed:
remove lock file

terezka committed:
Update example

terezka committed:
Fix missing import in tests

terezka committed:
Add missing deps

terezka committed:
Un wrap maybe's

In the newer version of Servant.Foreign, header args and query args are wrapped in maybe's. This commit adjusts accordingly.

terezka committed:
Add note

stoeffel commented on examples/e2e-tests/elm/Generated/Api.elm:
```diff
@@ -61,8 +64,8 @@ decodeResponseWithArgs =
 
-getIp : Http.Request (Http.Response (Response))
-getIp =
+getIp : (Result (Maybe (Http.Metadata, String), Http.Error) (Response) -> msg) -> Cmd msg
```
Check in the code that is generated for tutorials that this is indeed the right signature. I guess these e2e tests didn't run for a while.

stoeffel commented on examples/e2e-tests/elm/Generated/Api.elm:
```diff
@@ -76,20 +79,60 @@ getIp =
-getStatus204 : Http.Request (Http.Response (NoContent))
-getStatus204 =
+
+getIpSimulated : (Result (Maybe (Http.Metadata, String), Http.Error) (Response) -> msg) -> ProgramTest.SimulatedEffect msg
```
I think we want this function to take a `String` from https://github.com/NoRedInk/servant-elm/pull/10 so that we can use that as the the `body_`, right?

terezka replied:
(We talk IRL about how this is not necessary)

terezka committed:
Aligh lts with monorepo

terezka committed:
nvm that

allenap commented:
I had a go at building and testing this in a Stackage 12.26 environment using Nix in https://github.com/NoRedInk/servant-elm/pull/12 but it's WIP. There is probably an easier way though, perhaps/probably with Stack itself.
